src/main/java/randoop/reflection/OperationModel.java

```diff
@@ -297,24 +303,56 @@ public void addClassLiterals(
    * @return a map from each class type to the set of methods/constructors in it
    * @throws OperationParseException if a method signature cannot be parsed
    */
-  public static MultiMap<Type, TypedOperation> readOperations(Path file, boolean onlyMethods)
+  public static MultiMap<Type, TypedClassOperation> readOperations(Path file, boolean onlyMethods)
       throws OperationParseException {
-    MultiMap<Type, TypedOperation> observerMap = new MultiMap<>();
     if (file != null) {
       try (EntryReader er = new EntryReader(file, "(//|#).*$", null)) {
-        for (String line : er) {
-          String sig = line.trim();
-          TypedClassOperation operation =
-              signatureToOperation(
-                  sig, VisibilityPredicate.IS_ANY, new EverythingAllowedPredicate());
-          observerMap.add(operation.getDeclaringType(), operation);
-        }
+        return readOperations(er, onlyMethods);
+
       } catch (IOException e) {
         String message = String.format("Error while reading file %s: %s%n", file, e.getMessage());
         throw new RandoopUsageError(message, e);
       }
     }
-    return observerMap;
+    return new MultiMap<>();
+  }
+
+  /**
+   * Returns operations read from the given EntryReader.
+   *
+   * @param er the EntryReader to read from.
+   * @param onlyMethods if true, throw an exception if a constructor is read
+   * @return contents of the file, as a map of operations
+   */
+  public static MultiMap<Type, TypedClassOperation> readOperations(
+      EntryReader er, boolean onlyMethods) {
+    MultiMap<Type, TypedClassOperation> operationsMap = new MultiMap<>();
+    for (String line : er) {
+      String sig = line.trim();
+      TypedClassOperation operation =
+          signatureToOperation(sig, VisibilityPredicate.IS_ANY, new EverythingAllowedPredicate());
+      operationsMap.add(operation.getDeclaringType(), operation);
+    }
+    return operationsMap;
+  }
+
+  /**
+   * Returns operations read from the given stream.
+   *
+   * @param is the stream from which to read
+   * @param filename the file name to use in diagnostic messages
+   * @param onlyMethods if true, throw an exception if a constructor is read
+   * @return contents of the file, as a map of operations
+   */
+  public static MultiMap<Type, TypedClassOperation> readOperationsFromStream(
```

Comment:
The other two versions are called `readOperations`.  It's inconsistent that just this one has a distinct name.